test/sdk/MathTest.ooc

```diff
@@ -55,47 +55,75 @@ MathTest: class extends Fixture {
-			
+
 			expect(10.0f clamp(8.0f, 14.0f), is equal to(10.0f) within(floatTolerance))
 			expect(7.0f clamp(9.0f, 11.0f), is equal to(9.0f) within(floatTolerance))
 			expect(13.0f clamp(9.0f, 11.0f), is equal to(11.0f) within(floatTolerance))
 			expect((-2.0f) clamp(-1.9f, 5.0f), is equal to(-1.9f) within(floatTolerance)) // (-2.0f) with parentheses because of bug in rock
-			
+
 			expect(1.999f equals(2.0f), is false)
 			expect(1.99999f equals(2.0f), is true)
-			
+
 			expect(Float absolute(-2.3f), is equal to(2.3f) within(floatTolerance))
 			expect(Float absolute(2.3f), is equal to(2.3f) within(floatTolerance))
-			
+
 			expect(Float sign(-2.3f), is equal to(-1.0f) within(floatTolerance))
 			expect(Float sign(2.3f), is equal to(1.0f) within(floatTolerance))
-			
+
 			expect(Float maximum(-1.2f, -1.1f), is equal to(-1.1f) within(floatTolerance))
 			expect(Float maximum(-1.2f, 0.f), is equal to(0.f) within(floatTolerance))
 			expect(Float minimum(-1.2f, -1.1f), is equal to(-1.2f) within(floatTolerance))
 			expect(Float minimum(-1.2f, 0.f), is equal to(-1.2f) within(floatTolerance))
-			
+
 			expect(10.f squared(), is equal to(100.f) within(floatTolerance))
-			
+
 			expect(Float linearInterpolation(2.0f, 5.0f, 0.5f), is equal to(3.5f) within(floatTolerance))
 			expect(Float linearInterpolation(-9.0f, 1.0f, 0.1f), is equal to(-8.f) within(floatTolerance))
+
+			nearZero := (0.1f + 0.1f + 0.1f) - 0.3f
+			expect(nearZero equals(0.0), is true)
+			expect(nearZero lessOrEqual(0.0), is true)
+			expect(nearZero greaterOrEqual(0.0), is true)
+			expect(nearZero greaterThan(0.0), is false)
+			expect(nearZero lessThan(0.0), is false)
+
+			nearZero = 0.3f - (0.1f + 0.1f + 0.1f)
+			expect(nearZero equals(0.0), is true)
+			expect(nearZero lessOrEqual(0.0), is true)
+			expect(nearZero greaterOrEqual(0.0), is true)
+			expect(nearZero greaterThan(0.0), is false)
+			expect(nearZero lessThan(0.0), is false)
 		})
 		this add("Double", func {
 			expect(Double toRadians(0.0), is equal to(0.0) within(doubleTolerance))
 			expect(Double toRadians(45.0), is equal to(0.78539) within(doubleTolerance))
 			expect(Double toDegrees(3.1415926535), is equal to(180.0) within(doubleTolerance))
-			
+
 			expect(10.0 clamp(8.0, 14.0), is equal to(10.0) within(doubleTolerance))
 			expect(7.0 clamp(9.0, 11.0), is equal to(9.0) within(doubleTolerance))
 			expect(13.0 clamp(9.0, 11.0), is equal to(11.0) within(doubleTolerance))
 			expect((-2.0) clamp(-1.9, 5.0), is equal to(-1.9) within(doubleTolerance)) // (-2.0) with parentheses because of bug in rock
-			
+
 			expect(1.999 equals(2.0), is false)
 			expect(1.99999 equals(2.0), is true)
+
+			nearZero := (0.1 + 0.1 + 0.1) - 0.3
+			expect(nearZero equals(0.0), is true)
+			expect(nearZero lessOrEqual(0.0), is true)
+			expect(nearZero greaterOrEqual(0.0), is true)
+			expect(nearZero greaterThan(0.0), is false)
+			expect(nearZero lessThan(0.0), is false)
+
+			nearZero = 0.3 - (0.1 + 0.1 + 0.1)
```

Comment:
I wanted to test both negative and positive epsilon, that's why this variable is reassigned here.